tests/integration_tests/http_gateway_test.rs

```diff
@@ -0,0 +1,364 @@
+use crate::helpers::{
+    ExpectedResponse, StaticAsset, TestEnv, check_response, get_asset_canister_wasm,
+    get_large_assets_canister_wasm, install_canister, retry_async, upload_asset_to_asset_canister,
+};
+use anyhow::{Context, anyhow, bail};
+use candid::Principal;
+use http::{HeaderValue, Method, StatusCode};
+use reqwest::{Client, Request};
+use std::time::Duration;
+use tokio::runtime::Runtime;
+use tracing::info;
+use url::Url;
+
+const FETCH_ASSET_RETRY_TIMEOUT: Duration = Duration::from_secs(20);
+const FETCH_ASSET_RETRY_INTERVAL: Duration = Duration::from_secs(2);
+
+// Test scenario:
+// - deploy asset canister via pocket-ic interface
+// - upload some sample assets to the canister
+//   - root path: "/"
+//   - JS file: "/foo.js"
+//   - non-root path: "/a/b/c"
+//   - invalid data (broken certification): "/invalid_data.txt"
+// - retrieve asset via HTTP client through ic-gateway endpoint
+// - verify asset integrity
+
+pub fn basic_http_gateway_test(env: &TestEnv) -> anyhow::Result<()> {
+    info!("install asset canister ...");
+    let asset_canister_id = install_canister(
+        &env.pic,
+        Principal::anonymous(),
+        None,
+        get_asset_canister_wasm(),
+    );
+
+    info!("setup HTTP client ...");
+    let certified_domain = format!("{asset_canister_id}.{}", env.ic_gateway_domain);
+    let raw_domain = format!("{asset_canister_id}.raw.{}", env.ic_gateway_domain);
+
+    let http_client = Client::builder()
+        .resolve(certified_domain.as_str(), env.ic_gateway_addr)
+        .resolve(raw_domain.as_str(), env.ic_gateway_addr)
+        .build()
+        .context(anyhow!("failed to build http client"))?;
+
+    let test_cases = [
+        (
+            "certified root".to_string(),
+            StaticAsset {
```

Comment:
Maybe it would be shorter to add `#[derive(derive_new::new)]` to `StaticAsset` and then just create it with `StaticAsset::new()` avoiding all these struct field names repetitions?

Also by adding `#[new(into)]` to the fields we can avoid `.to_string()` boilerplate